### PR TITLE
Separate fetch and display

### PR DESCRIPTION
**Pull request by ajain-work**
Implemented following changes:

- Separated fetching and rendering. It allows to set data directly from other source e.g. if data is received by streaming.
- Meshes are colored by id.
- Image can be exported as html.
- Graphics can be overlayed over each other,

To set data from another source
---------------------------------------
```
pyvista_windows_manager.open_window(window_id)
active_window = pyvista_windows_manager.get_window(window_id)
active_window.post_object = mesh 
active_window.set_data(FieldDataType.Meshes ,data_from_another_source)  
pyvista_windows_manager.refresh_windows(session_id, [window_id])
```

To Overlay
------------
```
mesh1.display("w1")
contour1.display("w1",overlay=true)
```
Temperature contour overlayed over mesh
_____________________________________________
overlay

To Export as html
--------------------

```
set_config(blocking=True, set_view_on_display="isometric")
pyvista_windows_manager.open_window("w1")
active_window = pyvista_windows_manager.get_window("w1")
active_window.show_window = False 
active_window.post_object = mesh1
active_window.plot()
plotter= pyvista_windows_manager.get_plotter("w1")
plotter.export_html('manifold.html',  backend='panel')
```
html

### CODE REVIEW

File: src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py [hunk 1]
```diff
@@ -38,24 +48,68 @@ def __init__(self, id: str, post_object: GraphicsDefn):
...
+        self.fetch_data: bool = False
+        self.show_window: bool = True
         self.animate: bool = False
         self.close: bool = False
         self.refresh: bool = False
         self.update: bool = False
         self._visible: bool = False
         self._init_properties()
+        self._data = {}
+        self._colors = {
```
Comment (seanpearsonuk) [inline src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py:61]: @ajain-work do these keys have a practical purpose?

Comment (ajain-work) [reply src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py:61]: No. Currently we are coloring automatically. But these will be needed if we do manual coloring.

File: src/ansys/fluent/visualization/pyvista/pyvista_objects.py [hunk 2]
```diff
@@ -59,16 +59,21 @@ class Mesh(MeshDefn):
...
+    def display(self, window_id: Optional[str] = None, overlay: Optional[bool] = False):
         """Display mesh graphics.
 
         Parameters
         ----------
         window_id : str, optional
             Window ID. If an ID is not specified, a unique ID is used.
             The default is ``None``.
+        overlay : bool, optional
+            If set to ``True``, graphics will be overlaid over existing graphics.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_objects.py:71]:
```suggestion
            Whether to overlay graphics over existing graphics.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_objects.py [hunk 3]
```diff
@@ -86,16 +91,21 @@ class Pathlines(PathlinesDefn):
...
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_objects.py:103]:
```suggestion
            Whether to overlay graphics over existing graphics.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_objects.py [hunk 4]
```diff
@@ -124,16 +134,21 @@ class Surface(SurfaceDefn):
...
+    def display(self, window_id: Optional[str] = None, overlay: Optional[bool] = False):
         """Display surface graphics.
 
         Parameters
         ----------
         window_id : str, optional
             Window ID. If an ID is not specified, a unique ID is used.
             The default is ``None``.
+        overlay : bool, optional
+            If set to ``True``, graphics will be overlaid over existing graphics.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_objects.py:146]:
```suggestion
            Whether to overlay graphics over existing graphics.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_objects.py [hunk 5]
```diff
@@ -160,16 +175,21 @@ class Contour(ContourDefn):
...
+    def display(self, window_id: Optional[str] = None, overlay: Optional[bool] = False):
         """Display contour graphics.
 
         Parameters
         ----------
         window_id : str, optional
             Window ID. If an ID is not specified, a unique ID is used.
             The default is ``None``.
+        overlay : bool, optional
+            If set to ``True``, graphics will be overlaid over existing graphics.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_objects.py:187]:
```suggestion
            Whether to overlay graphics over existing graphics.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_objects.py [hunk 6]
```diff
@@ -197,13 +217,18 @@ class Vector(VectorDefn):
...
+    def display(self, window_id: Optional[str] = None, overlay: Optional[bool] = False):
         """Display vector graphics.
 
         Parameters
         ----------
         window_id : str, optional
             Window ID. If an ID is not specified, a unique ID is used.
             The default is ``None``.
+        overlay : bool, optional
+            If set to ``True``, graphics will be overlaid over existing graphics.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_objects.py:229]:
```suggestion
            Whether to overlay graphics over existing graphics.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py [hunk 7]
```diff
@@ -465,7 +579,10 @@ def plot(self, object: GraphicsDefn, window_id: Optional[str] = None) -> None:
         window_id : str, optional
             Window ID for the plot. The default is ``None``, in which
             case a unique ID is assigned.
-
+        fetch_data : bool, optional
+            If set to `True`,data will always be fetched.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py:583]:
```suggestion
            Whether to fetch data. The default is ``False``.
```

File: src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py [hunk 8]
```diff
@@ -465,7 +579,10 @@ def plot(self, object: GraphicsDefn, window_id: Optional[str] = None) -> None:
...
+        overlay : bool, optional
+            If set to `True`, new graphics will be overlaid over existing.
```
Comment (PipKat) [inline src/ansys/fluent/visualization/pyvista/pyvista_windows_manager.py:585]:
```suggestion
            Whether to overlay graphics over existing graphics. The default is ``False``.
```

**Review by prmukherj — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Pull request merged**